rustls/src/client/hs.rs

```diff
@@ -108,10 +108,11 @@ pub(super) fn start_handshake(
 
     let mut resuming = find_session(&server_name, &config, cx);
 
-    let key_share = if config.supports_version(ProtocolVersion::TLSv1_3) {
-        Some(tls13::initial_key_share(&config, &server_name)?)
+    let (group, key_share) = if config.supports_version(ProtocolVersion::TLSv1_3) {
+        let group_and_keyshare = tls13::initial_key_share(&config, &server_name)?;
+        (Some(group_and_keyshare.0), Some(group_and_keyshare.1))
```

Comment:
This is a bit gross but I couldn't think of a nicer way to transform a `(&'static dyn SupportedKxGroup, Box<dyn ActiveKeyExchange>)` into a `(Option<&'static dyn SupportedKxGroup>, Option<Box<dyn ActiveKeyExchange>>)`.

Making this conditional emit an optional tuple instead of two optional values seemed like it made the rest of the code uglier.

Alternative suggestions welcome.